server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java

```diff
@@ -1281,6 +1281,10 @@ private Optional<CompactionInfo> reserveFilesForCompaction(CompactionServiceId s
 
       cInfo.localHelper = this.chelper;
       cInfo.localCompactionCfg = this.compactionConfig;
+      // don't include execution hints, which may have been previously set, for system compactions
+      if (job.getKind() == CompactionKind.SYSTEM) {
+        cInfo.localCompactionCfg.setExecutionHints(Collections.emptyMap());
```

Comment:
`cInfo.localCompactionCfg` is a reference to `this.compactionConfig` so changing it will change `this.compactionConfig` which could cause bugs with user compactions. 

`cInfo.localCompactionCfg` really only needs to be logged for user compactions.  One way to achieve this is to only set `cInfo.localCompactionCfg` for user compactions and have it be null otherwise.  This would log null for the config then.

Reply:
Ah okay wasn't sure if other config was still needed in the logs for SYSTEM compactions. Addressed

Reply:
Also manually verified that this change works